selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/EditorBehaviorDuringUnexpectedWsAgentStopTest.java

```diff
@@ -0,0 +1,152 @@
+  private static final String TEXT_FOR_TYPING = "text for checking";
+  private static final String FIND_WS_AGENT_PID_COMMAND = "jps | grep Bootstrap";
+  private static final String FILE_NAME = "Aclass";
+  private static final String PATH_TO_CHECKING_FILE =
+      PROJECT_NAME + "/src/main/java/che.eclipse.sample";
+  private static final String EXPECTED_EDITOR_TEXT =
+      "/*\n"
+          + " * Copyright (c) 2012-2018 Red Hat, Inc.\n"
+          + " * All rights reserved. This program and the accompanying materials\n"
+          + " * are made available under the terms of the Eclipse Public License v1.0\n"
+          + " * which accompanies this distribution, and is available at\n"
+          + " * http://www.eclipse.org/legal/epl-v10.html\n"
+          + " *\n"
+          + " * Contributors:\n"
+          + " *   Red Hat, Inc. - initial API and implementation\n"
+          + " */\n"
+          + "package che.eclipse.sample;\n"
+          + "\n"
+          + "public class Aclass {\n"
+          + "}\n";
+
+  @Inject private ProjectExplorer projectExplorer;
+  @Inject private CodenvyEditor editor;
+  @Inject private Ide ide;
+  @Inject private TestWorkspace testWorkspace;
+  @Inject private TestProjectServiceClient testProjectServiceClient;
+  @Inject private DefaultTestUser defaultTestUser;
+  @Inject private DockerCliCommandExecutor dockerCliCommandExecutor;
+  @Inject private TestWorkspaceServiceClient testWorkspaceServiceClient;
+  @Inject private NotificationsPopupPanel notificationsPopupPanel;
+  @Inject private ToastLoader toastLoader;
+
+  @BeforeClass
+  public void prepare() throws Exception {
+    Path projectPath =
+        Paths.get(getClass().getResource("/projects/default-spring-project").toURI());
+    testProjectServiceClient.importProject(
+        testWorkspace.getId(), projectPath, PROJECT_NAME, ProjectTemplates.MAVEN_SPRING);
+
+    ide.open(testWorkspace);
+  }
+
+  @Test
+  public void editorShouldBeInReadonlyModeAfterUnexpectedWsAgentStopping() throws Exception {
+    goToFileAndCheckText();
+
+    waitWorkspaceRunningStatus();
+
+    executeKillWsAgentCommand();
+
+    // check that ws agent has stopped and current editor tab is displayed
+    notificationsPopupPanel.waitExpectedMessageOnProgressPanelAndClosed(
+        EXPECTED_POPUP_MESSAGE, 180);
+    toastLoader.waitToastLoaderIsOpen();
+    toastLoader.waitExpectedTextInToastLoader("Workspace agent is not running");
+    editor.waitTabIsPresent(FILE_NAME);
+    editor.waitTextIntoEditor(EXPECTED_EDITOR_TEXT);
+
+    // check that editor is in read only mode
+    editor.waitActive();
+    editor.typeTextIntoEditor(TEXT_FOR_TYPING);
+    editor.waitTextIntoEditor(EXPECTED_EDITOR_TEXT);
+
+    // check ws agent restoring
+    toastLoader.clickOnToastLoaderButton("Restart");
+    toastLoader.waitExpectedTextInToastLoader("Starting workspace runtime...");
+    waitWorkspaceRunningStatus();
+    goToFileAndCheckText();
```

Comment:
IMHO it may be reasonable to check that editor switched from read-only to normal mode after the workspace had been restored.